sale_contact_type/models/res_partner.py

```diff
@@ -0,0 +1,19 @@
+# Copyright 2019 Alexandre Díaz
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, models, fields
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    type = fields.Selection(selection_add=[('ordering', 'Ordering address')])
+    has_ordering_contact_child = fields.Boolean(
+        compute='_has_ordering_contact_child',
+        help='Technical field for use in views', store=True)
+
+    @api.depends('child_ids.type')
+    def _has_ordering_contact_child(self):
+        for record in self:
+            record.has_ordering_contact_child = any(record.child_ids.filtered(
```

Comment:
Can you optimize this with `read_group`? I think you can group by `parent_id` and by `type`, and assume all yes unless there's a group for `ordering` (and has parent).

Reply:
Moreover, >= v13, you need to initialize all records with initial value as they are not by default before the compute function.